code/modules/mob/living/carbon/human/species_types.dm

```diff
@@ -622,6 +740,229 @@
 			active = null
 	..()
 
+//Immune to physical bullets and resistant to brute, but very vulnerable to burn damage. Dusts on death.
+/datum/species/golem/sand
+	name = "Sand Golem"
+	id = "sand"
+	fixed_mut_color = "ffdc8f"
+	meat = /obj/item/weapon/ore/glass //this is sand
+	armor = 0
+	burnmod = 3 //melts easily
+	brutemod = 0.25
+	info_text = "As a <span class='danger'>Sand Golem</span>, you are immune to physical bullets and take very little brute damage, but are extremely vulnerable to burn damage. You will also turn to sand when dying, preventing any form of recovery."
+
+/datum/species/golem/sand/spec_death(gibbed, mob/living/carbon/human/H)
+	H.visible_message("<span class='danger'>[H] turns into a pile of sand!</span>")
+	for(var/obj/item/W in H)
+		H.unEquip(W)
+	for(var/i=1, i <= rand(3,5), i++)
+		new /obj/item/weapon/ore/glass(get_turf(H))
+	qdel(H)
+
+/datum/species/golem/sand/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	if(!(P.original == H && P.firer == H))
+		if(istype(P, /obj/item/projectile/bullet) || istype(P, /obj/item/projectile/kinetic))
```

Comment:
This should check the projectile's damage flag(the `flag` var) instead of types, ie; `if(P.flag == "bullet" || P.flag == "bomb")`